pkgs/by-name/pl/plasticscm-client-gui/package.nix

```diff
@@ -0,0 +1,102 @@
+{
+  buildFHSEnv,
+  extraLibs ? _: [ ],
+  extraPkgs ? _: [ ],
+  imagemagick,
+  lib,
+  makeWrapper,
+  plasticscm-client-core-unwrapped,
+  plasticscm-client-gui-unwrapped,
+  plasticscm-theme,
+}:
+assert plasticscm-client-gui-unwrapped.version == plasticscm-client-core-unwrapped.version;
+assert plasticscm-client-gui-unwrapped.version == plasticscm-theme.version;
+buildFHSEnv {
+  pname = "plasticscm-client-gui";
+  inherit (plasticscm-client-gui-unwrapped) version meta;
+
+  runScript = "";
+
+  targetPkgs =
+    pkgs:
+    with pkgs;
+    [
+      plasticscm-client-gui-unwrapped
+
+      # Dependencies from the Debian package
+      plasticscm-client-core-unwrapped
+
+      # Undocumented dependencies discovered from testing
+      fontconfig
+    ]
+    ++ extraPkgs pkgs;
+
+  multiPkgs =
+    pkgs:
+    with pkgs;
+    [
+      # Dependencies from the Debian package
+      glibc.out
+      libgcc.lib
+      krb5.lib
+      lttng-ust.out
+      openssl_3.out
+      icu76
+      plasticscm-theme
+
+      # Transitive dependencies from the Debian package
+      libidn2.out
+      libunistring
+      e2fsprogs.out
+      keyutils.lib
+      numactl.out
+      libz
+
+      # Undocumented dependencies discovered from testing
+      gtk3
+      xorg.libICE
+      xorg.libSM
+      xorg.libX11
+      xorg.libXcursor
+      xorg.libXi
+    ]
+    ++ extraLibs pkgs;
+
+  nativeBuildInputs = [
+    makeWrapper
+    imagemagick
+  ];
+
+  extraBuildCommands = ''
+    for app in linplasticx lingluonx; do
+      cp --remove-destination $(readlink $out/opt/plasticscm5/client/$app) $out/opt/plasticscm5/client/$app
+    done
+  '';
+
+  extraInstallCommands = ''
+    mv $out/bin/$pname $out/bin/.$pname-fhsenv
+
+    for app in plasticgui semanticmergetool gtkmergetool gluon; do
+      makeWrapper $out/bin/.$pname-fhsenv $out/bin/$app \
```

Comment:
Please hardcode pname instead.

Reply:
We should never rely on pname like this. It is an anti pattern we are trying to get rid of.

Reply:
I've hardcoded the `pname`, but I still don't quite understand the reasoning here. Is there a wider discussion on why this is bad *for this particular case*?